src/main/java/sirius/biz/tenants/TenantController.java

```diff
@@ -369,7 +370,14 @@ private String determineOriginalTenantId(WebContext ctx) {
     @Routed("/tenants/select/:1")
     public void selectTenant(final WebContext ctx, String id) {
         if ("main".equals(id) || Strings.areEqual(determineOriginalTenantId(ctx), id)) {
-            auditLog.neutral("AuditLog.switchedToMainTenant").causedByCurrentUser().forCurrentUser().log();
+            UserAccount<?, ?> account = UserContext.getCurrentUser().as(UserAccount.class);
+            auditLog.neutral("AuditLog.switchedToMainTenant")
+                    .hideFromUser()
+                    .causedByUser(account.getUniqueName(), account.getUserAccountData().getLogin().getUsername())
+                    .forUser(account.getUniqueName(), account.getUserAccountData().getLogin().getUsername())
```

Comment:
why not `forCurrentUser()` and `causedByCurrentUser()`?

Reply:
because the user is a malformed beast here which sill contains the wrong tenant...

Reply:
`UserContext.getCurrentUser().as(UserAccount.class).getTenant().getTenantData().getName()` should be exactly the same thing as `user.getTenantName()`, because the beast with the wrong tenant also sets the tenantName of the UserInfo, or am I missing something?

`UserContext.getCurrentUser().as(UserAccount.class)` is also returning the UserAccount with the "not original" tenant, or am I wrong?